src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -137,8 +137,18 @@ public void LoadPeers()
             {
                 // Ensure that any address already in store is mapped.
                 peer.Endpoint = peer.Endpoint.MapToIpv6();
-                
-                this.peers.TryAdd(peer.Endpoint, peer);
+
+                // If no longer banned reset ban details.
+                if (peer.BanUntil.HasValue && peer.BanUntil < this.dateTimeProvider.GetUtcNow())
```

Comment:
So we only clear this list on startup, that's fine I guess as it indirectly means we clear it on shutdown.